packages/ui/src/components/va-stepper/VaStepper.vue

```diff
@@ -0,0 +1,297 @@
+          :name="`step-${i}`"
+          v-bind="{
+            ...stepControls,
+            step,
+            isActive: $props.modelValue === i,
+            isCompleted: $props.modelValue > i,
+          }"
+        >
+          <li
+            class="va-stepper__step"
+            :class="{
+              'va-stepper__step--active': $props.modelValue >= i,
+              'va-stepper__step--disabled': step.disabled || isNextStepDisabled(i),
+              'va-stepper__step--navigation-disabled': $props.navigationDisabled,
+            }"
+            @click="!$props.navigationDisabled && stepControls.setStep(i)"
+          >
+
+            <div class="va-stepper__step__icon">
+              <va-icon
+                v-if="step.icon"
+                :name="step.icon"
+                size="1.3rem"
+              />
+              <span v-else>
+                {{ i + 1 }}
+              </span>
+            </div>
+
+            <span>
+              {{ step.label }}
+            </span>
+          </li>
+        </slot>
+      </template>
+    </ol>
+    <div
+      class="va-stepper__content-wrapper"
+      :class="{ 'va-stepper__content-wrapper--vertical': $props.vertical }"
+    >
+      <template
+        v-for="step, i in $props.steps"
+        :key="i"
+      >
+        <div
+          class="va-stepper__content"
+          v-if="$slots[`content-${i}`] && $props.modelValue === i"
+        >
+          <slot
+            :name="`content-${i}`"
+            v-bind="{ ...stepControls, step }"
+          />
+        </div>
+      </template>
+      <div class="va-stepper__controls">
+        <template v-if="!noControls">
+          <VaButton
+            preset="primary"
+            :disabled="$props.modelValue <= 0"
+            @click="stepControls.prevStep()"
+          >
+            Back
+          </VaButton>
+          <VaButton
+            v-if="isNextButtonVisible"
+            @click="stepControls.nextStep()"
+            :disabled="$props.nextDisabled"
+          >
+            Next
+          </VaButton>
+        </template>
+        <slot
+          name="controls"
+          v-bind="stepControls"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import { defineComponent, PropType, computed } from 'vue'
+import { VaIcon } from '../va-icon'
+import { VaButton } from '../va-button'
+import { useColors } from '../../composables'
+
+type Step = {
+  label: string,
+  icon?: string,
+  disabled?: boolean,
+}
+
+export default defineComponent({
+  name: 'VaStepper',
+  components: { VaIcon, VaButton },
+  props: {
+    modelValue: { type: Number, required: true, default: 0 },
+    steps: {
+      type: Array as PropType<Step[]>,
+      default: () => [],
+      required: true,
+    },
+    color: { type: String, default: 'primary' },
+    vertical: { type: Boolean, default: false },
+    navigationDisabled: { type: Boolean, default: false },
+    noControls: { type: Boolean, default: false },
```

Comment:
Last my pr Asva asked not to name boolean props like `noSmthng`. Here we can rename it to `controlsDisabled`.